test/unit/steps/do-query/do-get.js

```diff
@@ -0,0 +1,108 @@
+import {expect} from "chai";
+import ResourceTypeRegistry from "../../../../src/ResourceTypeRegistry";
+import doGet from "../../../../src/steps/do-query/do-get";
+
+class TestAdapter {
+  static getReferencedType(model, path) {
+    if (path === "some-relationship") {
+      return "dasherize-off-type";
+    }
+    return "dasherize-on-type";
+  }
+  find(type, idOrIds, fields, sorts, filters, includes) {
+    return {
+      then() {
+        return { type, idOrIds, fields, sorts, filters, includes };
+      }
+    };
+  }
+}
+
+describe("GET requests", () => {
+
+  let registry;
+  let dbAdapter = new TestAdapter();
+
+  before(() => {
+    registry = new ResourceTypeRegistry();
+    registry.type("dasherize-on-type", { dbAdapter: dbAdapter });
+    registry.type("dasherize-off-type", {
+      dbAdapter: dbAdapter,
+      behaviors: {
+        dasherizeOutput: { enabled: false }
+      }
+    });
+  });
+
+  describe("parsing query params", () => {
+    it("should camelize fields param according to type behavior", () => {
+      let request = {
+        type: "dasherize-on-type",
+        queryParams: {
+          fields: {
+            "dasherize-on-type": "a-dasherized-field",
+            "dasherize-off-type": "another-dasherized-field"
+          }
+        }
+      };
+
+      let passedToAdapter = doGet(request, {}, registry);
+      expect(passedToAdapter.fields).to.deep.equal({
```

Comment:
Are these tests actually passing? I thought `doGet` returns a promise (that ultimately resolves to `undefined`, since it just has the side effect of setting primary/included on the response). If so, how are you checking `passedToAdapter.fields`?